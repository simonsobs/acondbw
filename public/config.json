{
  "graphqlHttp": "http://localhost:5001/graphql"
}
