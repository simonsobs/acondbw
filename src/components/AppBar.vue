<template>
  <v-app-bar app dense clipped-left>
    <v-app-bar-nav-icon @click="$emit('drawer')" />
    <v-toolbar-title>
      <router-link to="/" v-text="title" style="text-decoration: none; color: inherit;"></router-link>
    </v-toolbar-title>
    <v-spacer></v-spacer>
    <search-window></search-window>
    <v-spacer></v-spacer>
    <sign-in-button></sign-in-button>
    <v-spacer></v-spacer>
    <v-btn icon to="/about">
      <v-icon>mdi-information</v-icon>
    </v-btn>

    <v-menu left bottom offset-y open-on-hover :close-on-content-click="false">
      <template v-slot:activator="{ on }">
        <v-btn icon v-on="on">
          <v-icon>mdi-dots-vertical</v-icon>
        </v-btn>
      </template>

      <v-list>
        <v-list-item @click="$vuetify.theme.dark = !$vuetify.theme.dark">
          <v-list-item-icon>
            <v-icon>mdi-invert-colors</v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title>Toggle dark mode</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
        <v-list-item :href="graphiqlUrl" target="_blank">
          <v-list-item-icon>
            <v-icon>mdi-graphql</v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title>GraphiQL</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </v-menu>
  </v-app-bar>
</template>

<script>
import SearchWindow from "@/components/SearchWindow";
import SignInButton from "@/components/SignInButton";

export default {
  name: "AppBar",
  components: {
    SearchWindow,
    SignInButton
  },
  data: () => ({
    title: process.env.VUE_APP_TOOLBAR_TITLE,
    graphiqlUrl: process.env.VUE_APP_GRAPHQL_HTTP
  }),
  created() {
    this.$vuetify.theme.dark = false;
  }
};
</script>
