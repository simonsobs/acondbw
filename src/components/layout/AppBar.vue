<template>
  <v-app-bar app dense clipped-left class="primary on-primary--text">
    <!-- <v-tooltip bottom open-delay="800">
      <template v-slot:activator="{ on }">
        <v-btn text icon exact v-on="on" @click="$router.go(-1)">
          <v-icon>mdi-arrow-left</v-icon>
        </v-btn>
      </template>
      <span>Back</span>
    </v-tooltip>-->
    <v-app-bar-nav-icon
      @click="$emit('drawer')"
      class="d-lg-none on-primary--text"
    />
    <v-toolbar-title>
      <router-link
        to="/"
        v-text="title"
        style="text-decoration: none; color: inherit"
      ></router-link>
    </v-toolbar-title>
    <v-spacer></v-spacer>
    <search-window class="d-none d-sm-flex"></search-window>
    <v-spacer></v-spacer>
    <user-menu-button></user-menu-button>
    <!-- <v-spacer></v-spacer>
    <v-btn icon to="/about">
      <v-icon>mdi-information</v-icon>
    </v-btn> -->

    <!-- <v-menu left bottom offset-y open-on-hover :close-on-content-click="false">
      <template v-slot:activator="{ on }">
        <v-btn icon v-on="on">
          <v-icon>mdi-dots-vertical</v-icon>
        </v-btn>
      </template>

      <v-list>
        <v-list-item @click="$vuetify.theme.dark = !$vuetify.theme.dark">
          <v-list-item-icon>
            <v-icon>mdi-invert-colors</v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title>Toggle dark mode</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
        <v-list-item :href="graphiqlUrl" target="_blank">
          <v-list-item-icon>
            <v-icon>mdi-graphql</v-icon>
          </v-list-item-icon>
          <v-list-item-content>
            <v-list-item-title>GraphiQL</v-list-item-title>
          </v-list-item-content>
        </v-list-item>
      </v-list>
    </v-menu> -->
  </v-app-bar>
</template>

<script lang="ts">
import Vue from "vue";
import SearchWindow from "@/components/utils/SearchWindow.vue";
import UserMenuButton from "@/components/auth/UserMenuButton.vue";

export default Vue.extend({
  name: "AppBar",
  components: {
    SearchWindow,
    UserMenuButton,
  },
  data: () => ({
    // graphiqlUrl: process.env.VUE_APP_GRAPHQL_HTTP,
    error: null,
  }),
  computed: {
    title() {
      return this.$store.state.webConfig.toolbarTitle || "";
    },
  },
  created() {
    this.$vuetify.theme.dark = false;
  },
});
</script>
