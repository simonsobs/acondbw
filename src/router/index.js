import Vue from "vue";
import VueRouter from "vue-router";
import VueMeta from "vue-meta";

import Frame from "@/components/layout/Frame";
import NullFrame from "@/components/layout/NullFrame";

import Home from "@/views/framework/Home.vue";
import Entry from "@/views/framework/Entry.vue";

import OAuthRedirect from "@/views/auth/OAuthRedirect.vue"

import SignIn from "@/views/auth/SignIn.vue";
import Auth from "@/views/auth/Auth.vue";
import SignInError from "@/views/auth/SignInError.vue";

import ProductTop from "@/views/product/ProductTop.vue";
import ProductList from "@/views/product/ProductList.vue";
import ProductItem from "@/views/product/ProductItem.vue";
import ProductAdd from "@/views/product/ProductAdd.vue";

import store from "@/store";

Vue.use(VueRouter);
Vue.use(VueMeta);

const routes = [
  {
    path: "/",
    name: "Entry",
    components: {
      default: Entry,
      frame: NullFrame,
    },
    beforeEnter: (to, from, next) => {
      const signedIn = !!store.state.auth.token;
      if (signedIn) {
        next({ name: "Dashboard" });
      } else {
        next();
      }
    },
  },
  {
    path: "/dashboard",
    name: "Dashboard",
    components: {
      default: Home,
      frame: Frame,
    },
    meta: { requiresAuth: true },
    beforeEnter: (to, from, next) => {
      const signedIn = !!store.state.auth.token;
      if (signedIn) {
        next();
      } else {
        next({ name: "Entry" });
      }
    },
  },
  {
    path: "/oauth-redirect",
    name: "OAuthRedirect",
    components: {
      default: OAuthRedirect,
      frame: NullFrame,
    },
  },
  {
    path: "/signin",
    name: "SignIn",
    components: {
      default: SignIn,
      frame: NullFrame,
    },
  },
  {
    path: "/auth",
    name: "Auth",
    components: {
      default: Auth,
      frame: NullFrame,
    },
  },
  {
    path: "/signin-error",
    name: "SignInError",
    components: {
      default: SignInError,
      frame: NullFrame,
    },
  },
  {
    path: "/admin/scratch",
    name: "AdminScratch",
    components: {
      default: () =>
        import(
          /* webpackChunkName: "admin" */ "@/views/admin-token/Scratch"
        ),
      frame: () =>
        import(
          /* webpackChunkName: "admin" */ "@/components/layout/FrameAdmin"
        ),
    },
  },
  {
    path: "/admin/token",
    name: "AdminAppToken",
    components: {
      default: () =>
        import(
          /* webpackChunkName: "admin" */ "@/views/admin-token/AdminAppToken"
        ),
      frame: () =>
        import(
          /* webpackChunkName: "admin" */ "@/components/layout/FrameAdmin"
        ),
    },
  },
  {
    path: "/admin/auth",
    name: "AdminAppAuth",
    components: {
      default: () =>
        import(
          /* webpackChunkName: "admin" */ "@/views/admin-token/AdminAppAuth"
        ),
        frame: () =>
        import(
          /* webpackChunkName: "admin" */ "@/components/layout/FrameAdmin"
        ),
    },
  },
  {
    path: "/admin/token-error",
    name: "AdminAppTokenError",
    components: {
      default: () =>
        import(
          /* webpackChunkName: "admin" */ "@/views/admin-token/AdminAppTokenError"
        ),
        frame: () =>
        import(
          /* webpackChunkName: "admin" */ "@/components/layout/FrameAdmin"
        ),
    },
  },
  {
    path: "/about",
    name: "About",
    component: () =>
      import(/* webpackChunkName: "about" */ "@/views/framework/About.vue"),
  },
  {
    path: "/add-product/:productTypeName",
    components: {
      default: ProductAdd,
      frame: NullFrame,
    },
    meta: { requiresAuth: true }
  },
  {
    path: "/product/:productTypeName",
    components: {
      default: ProductTop,
      frame: Frame,
    },
    meta: { requiresAuth: true },
    children: [
      {
        path: "",
        name: "ProductList",
        component: ProductList,
      },
      {
        path: ":name",
        name: "ProductItem",
        component: ProductItem,
      },
    ],
  },
  {
    path: "*",
    name: "NotFound",
    components: {
      default: () =>
        import(/* webpackChunkName: "error" */ "@/views/framework/NotFound.vue"),
        frame: NullFrame,
    },
  },
];

const router = new VueRouter({
  mode: "history",
  base: process.env.BASE_URL,
  routes,
});

router.beforeEach((to, from, next) => {
  const authRequired = to.matched.some((record) => record.meta.requiresAuth);
  const signedIn = !!store.state.auth.token;
  if (authRequired) {
    if (signedIn) {
      next();
    } else {
      next("/");
    }
  } else {
    next();
  }
});

export default router;
