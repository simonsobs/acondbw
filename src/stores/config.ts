import { ref, computed, watch, watchEffect } from "vue";
import { defineStore } from "pinia";
import { Client } from "@urql/vue";

const localStorageKey = "config";

import {
  WebConfigQuery,
  WebConfigDocument,
  SaveWebConfigMutation,
  SaveWebConfigDocument,
  SaveWebConfigMutationVariables,
} from "@/generated/graphql";

export interface VuetifyTheme {
  primary?: string;
  "on-primary"?: string;
  secondary?: string;
  "on-secondary"?: string;
  accent?: string;
  "on-accent"?: string;
  error?: string;
  "on-error"?: string;
  info?: string;
  "on-info"?: string;
  success?: string;
  "on-success"?: string;
  warning?: string;
  "on-warning"?: string;
}

export interface WebConfig extends VuetifyTheme {
  headTitle: string;
  toolbarTitle: string;
  devtoolLoadingstate: boolean;
  productCreationDialog: boolean;
  productUpdateDialog: boolean;
  productDeletionDialog: boolean;
}

function deepCopy<T>(data: T): T {
  return JSON.parse(JSON.stringify(data));
}

function _readFromLocalStorage() {
  const stringified = localStorage.getItem(localStorageKey);
  if (stringified === null) return null;
  let parsed: unknown;
  try {
    parsed = JSON.parse(stringified);
  } catch {
    return null;
  }
  if (typeof parsed !== "object") return null;
  return parsed;
}

function readFromLocalStorage() {
  const parsed = _readFromLocalStorage();
  if (parsed === null) localStorage.removeItem(localStorageKey);
  return parsed || {};
}

function writeToLocalStorage(data: unknown) {
  localStorage.setItem(localStorageKey, JSON.stringify(data));
}

export const useConfigStore = defineStore("config", () => {
  const error = ref<unknown | null>(null);

  const defaultConfig = ref<WebConfig>({
    headTitle: "",
    toolbarTitle: "",
    devtoolLoadingstate: false,
    productCreationDialog: false,
    productUpdateDialog: false,
    productDeletionDialog: false,
  });

  const configLocalStorage = ref({
    ...deepCopy(defaultConfig.value),
    ...readFromLocalStorage(),
  });
  writeToLocalStorage(configLocalStorage.value); // in case defaultConfig has added new properties

  const webConfig = ref(deepCopy(configLocalStorage.value));
  const webConfigJson = computed(() => JSON.stringify(webConfig.value));

  const configServerJson = ref<string | null>(null);
  const configServer = ref<WebConfig | null>(null);
  watch(configServerJson, (newValue) => {
    configServer.value = null;
    if (newValue === null) return;
    let parsed: unknown;
    try {
      parsed = JSON.parse(newValue);
    } catch (e) {
      error.value = e;
      return;
    }
    if (parsed === null) {
      error.value = new Error(
        "The config stored on the server is parsed to null"
      );
      return;
    }
    if (typeof parsed !== "object") {
      error.value = new Error(
        `The config stored on the server is not an object: ${parsed}`
      );
      return;
    }
    configServer.value = { ...deepCopy(defaultConfig.value), ...parsed };
    webConfig.value = deepCopy(configServer.value);
    configLocalStorage.value = deepCopy(webConfig.value);
    writeToLocalStorage(configLocalStorage.value);
  });

  const client = ref<Client | null>(null);
  watchEffect(async () => {
    await loadFromServer();
  });

  async function loadFromServer() {
    if (client.value === null) return;
    const { error: combinedError, data } = await client.value
      .query<WebConfigQuery>(WebConfigDocument, {})
      .toPromise();
    if (combinedError) {
      error.value = combinedError;
      return;
    }
    configServerJson.value = data?.webConfig?.json || null;
  }

  const saved = computed(
    () => !(webConfigJson.value === configServerJson.value)
  );

  function reset() {
    if (configServer.value) {
      webConfig.value = deepCopy(configServer.value);
    } else {
      webConfig.value = deepCopy(configLocalStorage.value);
    }
  }

  async function saveToServer() {
    if (client.value === null) return;
    const { error: combinedError, data } = await client.value
      .mutation<SaveWebConfigMutation, SaveWebConfigMutationVariables>(
        SaveWebConfigDocument,
        { json: webConfigJson.value }
      )
      .toPromise();
    if (combinedError) {
      error.value = combinedError;
      return;
    }
    await loadFromServer();
  }

  const vuetifyTheme = computed(() => {
    const theme_fields_base = [
      "primary",
      "secondary",
      "accent",
      "error",
      "info",
      "success",
      "warning",
    ];
    const theme_fields = [
      ...theme_fields_base,
      ...theme_fields_base.map((k) => `on-${k}`),
    ];
    return theme_fields
      .filter((e) => e in webConfig.value && webConfig.value[e])
      .reduce((a, e) => ({ ...a, ...{ [e]: webConfig.value[e] } }), {});
  });


  return {
    error,
    defaultConfig,
    configLocalStorage,
    webConfig,
    webConfigJson,
    configServer,
    configServerJson,
    client,
    saved,
    reset,
    loadFromServer,
    saveToServer,
    vuetifyTheme,
  };
});
