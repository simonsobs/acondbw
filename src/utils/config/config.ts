export interface Config {
  graphqlHttp: string;
}

export const defaultConfig = {};

export function validateConfig(config: Config) {
  if (typeof config.graphqlHttp !== "string")
    throw Error("graphqlHttp is not string");
  if (config?.graphqlHttp === "") throw Error("apiHttp is empty");
}
