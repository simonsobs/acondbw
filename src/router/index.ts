import Vue from "vue";
import VueRouter, { RouteConfig } from "vue-router";
import VueMeta from "vue-meta";

import { default as defaultPinia } from "@/stores";
import { useAuthStore } from "@/stores/auth";

import Frame from "@/components/layout/Frame.vue";
import NullFrame from "@/components/layout/NullFrame.vue";

import Home from "@/views/framework/Home.vue";
import Entry from "@/views/framework/Entry.vue";
import SearchTest from "@/views/framework/SearchTest.vue";

import OAuthRedirect from "@/views/auth/OAuthRedirect.vue";

import SignIn from "@/views/auth/SignIn.vue";
import Auth from "@/views/auth/Auth.vue";
import SignInError from "@/views/auth/SignInError.vue";

import ProductTop from "@/views/product/ProductTop.vue";
import ProductList from "@/views/product/ProductList.vue";
import ProductItem from "@/views/product/ProductItem.vue";
import ProductAdd from "@/views/product/ProductAdd.vue";

import ProductType from "@/views/admin/ProductType.vue";

const About = () => import("@/views/framework/About.vue");
const NotFound = () => import("@/views/framework/NotFound.vue");
const AccessDenied = () => import("@/views/framework/AccessDenied.vue");

const FrameAdmin = () =>
  import(/* webpackChunkName: "admin" */ "@/components/admin/FrameAdmin.vue");
const Version = () =>
  import(/* webpackChunkName: "admin" */ "@/views/admin/Version.vue");
const Log = () =>
  import(/* webpackChunkName: "admin" */ "@/views/admin/Log.vue");
const Config = () =>
  import(/* webpackChunkName: "admin" */ "@/views/admin/Config.vue");
const Theme = () =>
  import(/* webpackChunkName: "admin" */ "@/views/admin/Theme.vue");
const User = () =>
  import(/* webpackChunkName: "admin" */ "@/views/admin/User.vue");
const AdminAppToken = () =>
  import(
    /* webpackChunkName: "admin" */ "@/views/admin-token/AdminAppToken.vue"
  );
const AdminAppAuth = () =>
  import(
    /* webpackChunkName: "admin" */ "@/views/admin-token/AdminAppAuth.vue"
  );
const AdminAppTokenError = () =>
  import(
    /* webpackChunkName: "admin" */ "@/views/admin-token/AdminAppTokenError.vue"
  );

Vue.use(VueRouter);
Vue.use(VueMeta);

let pinia = defaultPinia;

function setPinia(val) {
  // because not sure how to monkey patch pinia for tests
  pinia = val;
}

function setDefaultPinia() {
  setPinia(defaultPinia);
}

const routes: Array<RouteConfig> = [
  {
    path: "/",
    name: "Entry",
    components: {
      default: Entry,
      frame: NullFrame,
    },
    beforeEnter: (to, from, next) => {
      const auth = useAuthStore(pinia);
      const signedIn = auth.isSignedIn;
      if (signedIn) {
        next({ name: "Dashboard" });
      } else {
        next();
      }
    },
  },
  {
    path: "/dashboard",
    name: "Dashboard",
    components: {
      default: Home,
      frame: Frame,
    },
    // meta: { requiresAuth: true },
    beforeEnter: (to, from, next) => {
      const auth = useAuthStore(pinia);
      const signedIn = auth.isSignedIn;
      if (signedIn) {
        next();
      } else {
        next({ name: "Entry" });
      }
    },
  },
  {
    path: "/search-test",
    name: "SearchTest",
    components: { default: SearchTest, frame: Frame },
    meta: { requiresAuth: true },
  },
  {
    path: "/oauth-redirect",
    name: "OAuthRedirect",
    components: { default: OAuthRedirect, frame: NullFrame },
  },
  {
    path: "/signin",
    name: "SignIn",
    components: { default: SignIn, frame: NullFrame },
  },
  {
    path: "/auth",
    name: "Auth",
    components: { default: Auth, frame: NullFrame },
  },
  {
    path: "/signin-error",
    name: "SignInError",
    components: { default: SignInError, frame: NullFrame },
  },
  {
    path: "/admin",
    name: "Admin",
    redirect: { name: "AdminVersion" },
  },
  {
    path: "/admin/product-type",
    name: "ProductType",
    components: { default: ProductType, frame: FrameAdmin },
    meta: { requiresAdmin: true },
  },
  {
    path: "/admin/versions",
    name: "AdminVersion",
    components: { default: Version, frame: FrameAdmin },
    meta: { requiresAdmin: true },
  },
  {
    path: "/admin/log",
    name: "AdminLog",
    components: { default: Log, frame: FrameAdmin },
    meta: { requiresAdmin: true },
  },
  {
    path: "/admin/config",
    name: "AdminConfig",
    components: { default: Config, frame: FrameAdmin },
    meta: { requiresAdmin: true },
  },
  {
    path: "/admin/theme",
    name: "AdminTheme",
    components: { default: Theme, frame: FrameAdmin },
    meta: { requiresAdmin: true },
  },
  {
    path: "/admin/users",
    name: "AdminUser",
    components: { default: User, frame: FrameAdmin },
    meta: { requiresAdmin: true },
  },
  {
    path: "/admin/token",
    name: "AdminAppToken",
    components: { default: AdminAppToken, frame: FrameAdmin },
  },
  {
    path: "/admin/auth",
    name: "AdminAppAuth",
    components: { default: AdminAppAuth, frame: FrameAdmin },
  },
  {
    path: "/admin/token-error",
    name: "AdminAppTokenError",
    components: { default: AdminAppTokenError, frame: FrameAdmin },
  },
  {
    path: "/about",
    name: "About",
    component: About,
  },
  {
    path: "/add-product/:productTypeName",
    name: "ProductAdd",
    components: { default: ProductAdd, frame: NullFrame },
    meta: { requiresAuth: true },
  },
  {
    path: "/product/:productTypeName",
    components: { default: ProductTop, frame: Frame },
    meta: { requiresAuth: true },
    children: [
      { path: "", name: "ProductList", component: ProductList },
      { path: ":name", name: "ProductItem", component: ProductItem },
    ],
  },
  {
    path: "/access-denied",
    name: "AccessDenied",
    components: { default: AccessDenied, frame: NullFrame },
  },
  {
    path: "*",
    name: "NotFound",
    components: { default: NotFound, frame: NullFrame },
  },
];

function createRouter() {
  const router = new VueRouter({
    mode: "history",
    base: process.env.BASE_URL,
    routes,
  });

  router.beforeEach((to, from, next) => {
    const auth = useAuthStore(pinia);

    const adminRequired = to.matched.some(
      (record) => record.meta.requiresAdmin
    );
    const isAdmin = auth.isAdmin;
    if (adminRequired && !isAdmin) {
      next({ name: "AccessDenied" });
      return;
    }

    const authRequired = to.matched.some((record) => record.meta.requiresAuth);
    const isSignedIn = auth.isSignedIn as boolean;
    if (authRequired && !isSignedIn) {
      next({ name: "AccessDenied" });
      return;
    }
    next();
  });

  return router;
}

const router = createRouter();

function checkAuthForCurrentRoute() {
  const authRequired = router.currentRoute.matched.some(
    (record) => record.meta.requiresAuth
  );
  const auth = useAuthStore(pinia);
  const signedIn = auth.isSignedIn as boolean;
  if (authRequired && !signedIn) {
    router.push("/");
  }
}

export { createRouter, checkAuthForCurrentRoute, setPinia, setDefaultPinia };
