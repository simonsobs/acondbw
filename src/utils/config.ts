import { storeToRefs } from "pinia";

import { useConfigStore } from "@/stores/config";

export function useConfig() {
  const store = useConfigStore();
  const { webConfig } = storeToRefs(store);
  const config = { config: webConfig };
  return config;
}
