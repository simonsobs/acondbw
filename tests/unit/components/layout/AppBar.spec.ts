import Vue, { ref } from "vue";
import VueRouter from "vue-router";
import { PiniaVuePlugin } from "pinia";
import Vuetify from "vuetify";
import { shallowMount, createLocalVue } from "@vue/test-utils";
import { createTestingPinia } from "@pinia/testing";

import { injectionKey } from "@/utils/config";

import AppBar from "@/components/layout/AppBar.vue";
import { createRouter } from "@/router";

Vue.use(Vuetify);
Vue.use(VueRouter);

describe("one", () => {
  // const ENV_ORG = process.env;

  let localVue: ReturnType<typeof createLocalVue>;
  let vuetify: Vuetify;
  let router: ReturnType<typeof createRouter>;

  function createWrapper(loading = false) {
    return shallowMount(AppBar, {
      localVue,
      vuetify,
      router,
      pinia: createTestingPinia(),
      provide: {
        [(injectionKey as any)]: { config: ref({ toolbarTitle: "Toolbar Title" }) },
      },
    });
  }

  beforeEach(() => {
    // process.env.VUE_APP_GRAPHQL_HTTP = "http://graphql.api:5000/graphql";
    localVue = createLocalVue();
    localVue.use(PiniaVuePlugin);
    vuetify = new Vuetify();
    router = createRouter();
  });

  afterEach(() => {
    // process.env = ENV_ORG;
  });

  it("match snapshot", async () => {
    const wrapper = createWrapper();
    await Vue.nextTick();
    expect(wrapper.html()).toMatchSnapshot();
  });
});
