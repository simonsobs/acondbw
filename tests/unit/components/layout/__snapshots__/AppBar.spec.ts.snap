// Vitest Snapshot v1

exports[`one > match snapshot 1`] = `
"<vuecomponent-stub tag=\\"header\\" dense=\\"true\\" extensionheight=\\"48\\" src=\\"\\" value=\\"true\\" app=\\"true\\" clippedleft=\\"true\\" class=\\"primary on-primary--text\\">
  <vuecomponent-stub class=\\"d-lg-none on-primary--text\\"></vuecomponent-stub>
  <vuecomponent-stub tag=\\"div\\">
    <router-link-stub to=\\"/\\" tag=\\"a\\" ariacurrentvalue=\\"page\\" event=\\"click\\" style=\\"text-decoration: none;\\"></router-link-stub>
  </vuecomponent-stub>
  <vuecomponent-stub tag=\\"div\\"></vuecomponent-stub>
  <searchwindow-stub class=\\"d-none d-sm-flex\\"></searchwindow-stub>
  <vuecomponent-stub tag=\\"div\\"></vuecomponent-stub>
  <usermenubutton-stub></usermenubutton-stub>
</vuecomponent-stub>"
`;
